pkg/storegateway/bucket.go

```diff
@@ -1192,6 +1191,28 @@ func (s *BucketStore) recordSeriesCallResult(safeStats *safeQueryStats) {
 	s.metrics.seriesHashCacheRequests.Add(float64(stats.seriesHashCacheRequests))
 	s.metrics.seriesHashCacheHits.Add(float64(stats.seriesHashCacheHits))
 
+	// Currently fineGrainedChunksCachingEnabled is always false. After it is configurable,
+	// these stats will be recorded by the new caching implementation.
+	if s.fineGrainedChunksCachingEnabled && s.maxSeriesPerBatch > 0 {
+		s.metrics.seriesDataFetched.WithLabelValues("chunks", "normal").Observe(float64(stats.chunksFetched))
+		s.metrics.seriesDataSizeFetched.WithLabelValues("chunks", "normal").Observe(float64(stats.chunksFetchedSizeSum))
+
+		s.metrics.seriesDataFetched.WithLabelValues("chunks", "refetch").Observe(float64(stats.chunksRefetched))
+		s.metrics.seriesDataSizeFetched.WithLabelValues("chunks", "refetch").Observe(float64(stats.chunksRefetchedSizeSum))
+
+		s.metrics.seriesDataTouched.WithLabelValues("chunks", "parsed").Observe(float64(stats.chunksParsed))
+		s.metrics.seriesDataSizeTouched.WithLabelValues("chunks", "parsed").Observe(float64(stats.chunksParsedSizeSum))
+
+		s.metrics.seriesDataTouched.WithLabelValues("chunks", "selected").Observe(float64(stats.chunksTouched))
```

Comment:
Have you considered renaming "selected" to "returned"? To my understanding these are the chunks returned because within the query min/max time range.